language/vm/vm_runtime/src/interpreter.rs

```diff
@@ -0,0 +1,1208 @@
+    where
+        VMResult<T>: From<Value>,
+        F: FnOnce(T, T) -> bool,
+    {
+        self.binop(|lhs, rhs| Some(Value::bool(f(lhs, rhs))))
+    }
+
+    /// Entry point for all global store operations (effectively opcodes).
+    ///
+    /// This performs common operation on the data store and then executes the specific
+    /// opcode.
+    fn global_data_op<F>(
+        &mut self,
+        address: AccountAddress,
+        idx: StructDefinitionIndex,
+        module: &LoadedModule,
+        op: F,
+    ) -> VMResult<AbstractMemorySize<GasCarrier>>
+    where
+        F: FnOnce(&mut Self, AccessPath, StructDef) -> VMResult<AbstractMemorySize<GasCarrier>>,
+    {
+        let ap = Self::make_access_path(module, idx, address);
+        if let Some(struct_def) =
+            self.module_cache
+                .resolve_struct_def(module, idx, &self.gas_meter)?
+        {
+            op(self, ap, struct_def)
+        } else {
+            Err(VMStatus::new(StatusCode::LINKER_ERROR))
+        }
+    }
+
+    /// BorrowGlobal (mutable and not) opcode.
+    fn borrow_global(
+        &mut self,
+        ap: AccessPath,
+        struct_def: StructDef,
+    ) -> VMResult<AbstractMemorySize<GasCarrier>> {
+        let global_ref = self.data_view.borrow_global(&ap, struct_def)?;
+        let size = global_ref.size();
+        self.operand_stack.push(Value::global_ref(global_ref))?;
+        Ok(size)
+    }
+
+    /// Exists opcode.
+    fn exists(
+        &mut self,
+        ap: AccessPath,
+        struct_def: StructDef,
+    ) -> VMResult<AbstractMemorySize<GasCarrier>> {
+        let (exists, mem_size) = self.data_view.resource_exists(&ap, struct_def)?;
+        self.operand_stack.push(Value::bool(exists))?;
+        Ok(mem_size)
+    }
+
+    /// MoveFrom opcode.
+    fn move_from(
+        &mut self,
+        ap: AccessPath,
+        struct_def: StructDef,
+    ) -> VMResult<AbstractMemorySize<GasCarrier>> {
+        let resource = self.data_view.move_resource_from(&ap, struct_def)?;
+        let size = resource.size();
+        self.operand_stack.push(resource)?;
+        Ok(size)
+    }
+
+    /// MoveToSender opcode.
+    fn move_to_sender(
+        &mut self,
+        ap: AccessPath,
+        struct_def: StructDef,
+    ) -> VMResult<AbstractMemorySize<GasCarrier>> {
+        let resource = self.operand_stack.pop_as::<Struct>()?;
+        let size = resource.size();
+        self.data_view.move_resource_to(&ap, struct_def, resource)?;
+        Ok(size)
+    }
+
+    /// Helper to create a resource storage key (`AccessPath`) for global storage operations.
+    fn make_access_path(
+        module: &impl ModuleAccess,
+        idx: StructDefinitionIndex,
+        address: AccountAddress,
+    ) -> AccessPath {
+        let struct_tag = resource_storage_key(module, idx);
+        create_access_path(&address, struct_tag)
+    }
+
+    //
+    // Create account opcode. This will all go away once we make create account a native function.
```

Comment:
Now I can see the mess here :D. It seems to me that we can inline all the three functions together as we won't reuse the code here?

Reply:
yeah I will do it

Reply:
I had to keep `save_account` because it's called from the other create account but inlined the other